domain/secret/state/state.go

```diff
@@ -1687,6 +1707,184 @@ ON CONFLICT(secret_id, unit_uuid) DO UPDATE SET
 		if err := tx.Query(ctx, insertStmt, consumer).Run(); err != nil {
 			return errors.Trace(err)
 		}
+
+		if err := st.markObsoleteRevisions(ctx, tx, uri); err != nil {
+			return errors.Annotatef(err, "marking obsolete revisions for secret %q", uri)
+		}
+
+		return nil
+	})
+	return errors.Trace(domain.CoerceError(err))
+}
+
+// GetSecretRemoteConsumer returns the secret consumer info from a cross model consumer
+// for the specified unit and secret.
+// If the secret does not exist, an error satisfying [secreterrors.SecretNotFound] is returned.
+// If there's not currently a consumer record for the secret, the latest revision is still returned,
+// along with an error satisfying [secreterrors.SecretConsumerNotFound].
+func (st State) GetSecretRemoteConsumer(ctx context.Context, uri *coresecrets.URI, unitName string) (*coresecrets.SecretConsumerMetadata, int, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, 0, errors.Trace(err)
+	}
+
+	consumer := secretRemoteUnitConsumer{
+		SecretID: uri.ID,
+		UnitID:   unitName,
```

Comment:
name or ID?

Reply:
We have the unit name passed in but the DDL uses unit_id.